wayland-backend/src/sys/client_impl/mod.rs

```diff
@@ -516,6 +516,37 @@ impl InnerBackend {
         }
     }
 
+    fn destroy_object_inner(&self, guard: &mut MutexGuard<ConnectionState>, id: &ObjectId) {
+        if let Some(ref alive) = id.id.alive {
+            let udata = unsafe {
+                Box::from_raw(ffi_dispatch!(
+                    wayland_client_handle(),
+                    wl_proxy_get_user_data,
+                    id.id.ptr
+                ) as *mut ProxyUserData)
+            };
+            unsafe {
+                ffi_dispatch!(
+                    wayland_client_handle(),
+                    wl_proxy_set_user_data,
+                    id.id.ptr,
+                    std::ptr::null_mut()
+                );
+            }
+            alive.store(false, Ordering::Release);
+            udata.data.destroyed(id.clone());
+        }
```

Comment:
This should check if `id.id.alive` is not already `False` (like if the user tries to destroy the same object twice), and return `Err(InvalidId)` if it is, to match the behavior of the rs backend.